generators/cmd/cli/cmd/show.go

```diff
@@ -0,0 +1,35 @@
+package cmd
+
+import (
+	"fmt"
+	"generators/pkg/parser"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var filename string
+var ShowCmd = &cobra.Command{
+	Use:   "show",
+	Short: "Show generated configuration.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(filename) == 0 {
```

Comment:
Currently this logic is not tested. Let's break this into a separate func that takes in a filename and a `io.Writer`- e.g. 

```go
func Write(filename string, w io.Writer) error {
...
}
```

This will also have the side effect of being able to be used by both `show` and `write` - one can pass in stdout as the writer while the other passes in another file.

Reply:
I am a little bit confused about this `Write` func and `WriteToDisk` func. For example if the command line is `tkn-gen show -f spec.yaml`, in the `Write`, I need to first parse the spec file then generate the Tekton Task and show the generated task as stdout. There is no need to call the `WriteToDisk`. But for the command `tkn-gen write -f spec.yaml`, in the `Write`, I need to call the `WriteToDisk` because it will write the output into a new yaml file. 

How could `Write` func be used by both `show` and `write`?

Reply:
Because both files and output streams (stdout, stderr) satisfy the [`io.Writer`](https://golang.org/pkg/io/#Writer) interface! :D 

Here's a concrete example: https://play.golang.org/p/8BWo0mWFmxE

Reply:
I change the `WriteToDisk` to:
```
func WriteToDisk(filename string, writer io.Writer) error {}
```
This will write the generated task from the input file to the io.Writer. It can be used by both `show` and `write` now. But only for one generated task. Next if I want to add support of the complete task and pipeline, it will change.